src/TurbDataReader.cpp

```diff
@@ -33,14 +33,14 @@ auto read_dataset(hid_t &file_id, char const *dataset_name) -> amrex::Table3D<do
 	}
 
 	// allocate array for dataset storage
-	auto *temp_data = new double[data_size];
+	std::unique_ptr<double[]> temp_data(new double[data_size]);
```

Comment:
This line causes the problem. Why did you change this? It will not work this way because when this function ends, the array will be deallocated, and any attempts to access the contents of `table` will cause a segmentation fault.

Reply:
`std::unique_ptr` deallocates the memory when it goes out of scope: https://en.cppreference.com/w/cpp/memory/unique_ptr

`amrex::Table3D` is just a wrapper around a raw pointer, so it can't keep it in scope: https://amrex-codes.github.io/amrex/doxygen/structamrex_1_1Table3D.html

Reply:
I realize clang-tidy is probably not happy with this line. However, as long as we *never* deallocate the memory allocated here, it does not cause a memory safety problem. It is not possible to clang-tidy to check this automatically, so it complains about allocating memory with a raw pointer like this.

The difficulty in fixing this in the way clang-tidy wants is that since `amrex::Table3D` does not allocate its own memory, we have to either i) allocate the memory before calling this function by creating, e.g., a `std::vector`, and then passing a reference to it to this function, or ii) return both an object that allocates its own memory (like `std::vector`, for example) *and* the `amrex::Table3D` object.